presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule.test;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.PlanPrinter;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.facebook.presto.sql.planner.assertions.PlanAssert.assertPlan;
+import static org.testng.Assert.fail;
+
+public class RuleAssert
+{
+    private final Metadata metadata;
+    private final Session session;
+    private final Rule rule;
+
+    private final PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
+
+    private Map<Symbol, Type> symbols;
+    private PlanNode plan;
+
+    public RuleAssert(Metadata metadata, Session session, Rule rule)
+    {
+        this.metadata = metadata;
+        this.session = session;
+        this.rule = rule;
+    }
+
+    public RuleAssert on(Function<PlanBuilder, PlanNode> planProvider)
+    {
+        PlanBuilder builder = new PlanBuilder(idAllocator);
+        plan = planProvider.apply(builder);
+        symbols = builder.getSymbols();
+        return this;
+    }
+
+    public void doesNotFire()
```

Comment:
I didn't want it to be confused with ".matches(<pattern>)", since that's about the *result* matching a pattern. If I named this "doesNotMatch" it would be about the *input* not matching.